src/TriggerBinding/SqlTriggerScaleMonitor.cs

```diff
@@ -0,0 +1,185 @@
+        private readonly int _maxChangesPerWorker;
+
+        public SqlTriggerScaleMonitor(string userFunctionId, string userTableName, string connectionString, int maxChangesPerWorker, ILogger logger)
+        {
+            _ = !string.IsNullOrEmpty(userFunctionId) ? true : throw new ArgumentNullException(userFunctionId);
+            _ = !string.IsNullOrEmpty(userTableName) ? true : throw new ArgumentNullException(userTableName);
+            this._userTable = new SqlObject(userTableName);
+            // Do not convert the scale-monitor ID to lower-case string since SQL table names can be case-sensitive
+            // depending on the collation of the current database.
+            this.Descriptor = new ScaleMonitorDescriptor($"{userFunctionId}-SqlTrigger-{this._userTable.FullName}", userFunctionId);
+            this._metricsProvider = new SqlTriggerMetricsProvider(connectionString, logger, this._userTable, userFunctionId);
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._maxChangesPerWorker = maxChangesPerWorker;
+        }
+
+        public ScaleMonitorDescriptor Descriptor { get; }
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+
+        }
+
+        public async Task<SqlTriggerMetrics> GetMetricsAsync()
+        {
+            return await this._metricsProvider.GetMetricsAsync().ConfigureAwait(false);
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            return this.GetScaleStatusWithTelemetry(context.WorkerCount, context.Metrics?.Cast<SqlTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<SqlTriggerMetrics> context)
+        {
+            return this.GetScaleStatusWithTelemetry(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusWithTelemetry(int workerCount, SqlTriggerMetrics[] metrics)
+        {
+            var status = new ScaleStatus
+            {
+                Vote = ScaleVote.None,
+            };
+
+            var properties = new Dictionary<TelemetryPropertyName, string>(this._telemetryProps)
+            {
+                [TelemetryPropertyName.ScaleRecommendation] = $"{status.Vote}",
+                [TelemetryPropertyName.TriggerMetrics] = metrics is null ? "null" : $"[{string.Join(", ", metrics.Select(metric => metric.UnprocessedChangeCount))}]",
+                [TelemetryPropertyName.WorkerCount] = $"{workerCount}",
+            };
+
+            try
+            {
+                status = this.GetScaleStatusCore(workerCount, metrics);
+
+                properties[TelemetryPropertyName.ScaleRecommendation] = $"{status.Vote}";
+                TelemetryInstance.TrackEvent(TelemetryEventName.GetScaleStatus, properties);
```

Comment:
Does this log go to some SQL internal logs?

Reply:
Sort of, it goes to our internal telemetry pipeline (I wouldn't call them "logs" though since it's more collection of statistics, for example in this case the number of times that a given type of scale recommendation was given)